modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -241,6 +241,14 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
   const rng = editor.selection.getRng();
   const newBlockName = Options.getForcedRootBlock(editor);
 
+  const isInRoot = rng.collapsed && rng.startContainer === editor.dom.getRoot();
```

Comment:
What about inside a `div` tag, rather than in the root? Does the bug not happen then?

Reply:
It does seem to occur there, I did not check for that. But I am wondering if the solution should not be slightly different for that as the div-block can be split using existing code whereas the body really can't be.

Reply:
you're right. New ticket please.